File: python/Binary Tree Inorder Traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def dfs(self, root, answer):
        if root:
            self.dfs(root.left, answer)
            answer.append(root.val)
            self.dfs(root.right, answer)
    
    def inorderTraversal(self, root):
        answer = []
        self.dfs(root, answer)
        return answer
        

File: cpp/Binary Tree Inorder Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> res;
    vector<int> inorderTraversal(TreeNode* root) {
        res.clear();
        helper(root);
        return res;
    }
    
    void helper(TreeNode *root) {
        if (root == NULL) {
            return;
        }
        helper(root->left);
        res.push_back(root->val);
        helper(root->right);
    }
};
